src/cluster.c

```diff
@@ -2070,3 +2070,13 @@ void resetClusterStats(void) {
 
     clusterSlotStatResetAll();
 }
+
+/* Invoked to clean up state after this node becomes a master. */
+void clusterBecameMaster(void) {
+    asmFinalizeMasterTask();
+}
+
+/* Invoked to clean up state after this node becomes a replica. */
+void clusterBecameReplica(void) {
+    clusterAsmCancel(NULL, "switching to replica");
+}
```

Comment:
These functions are a bit overlapping with existing ones, e.g. with `clusterPromoteSelfToMaster()`. I could reuse it but then I saw cluster_plugin.c has `clusterPromoteSelfToMaster()`. Maybe we can have these two distinct functions and ask plugin to call them.

Reply:
Question for Josh: are they calling replicationUnsetMaster() somewhere? How do they handle it?